controller.js

```diff
@@ -55,21 +58,43 @@ exports.getAllArticles = async (req, res, next) => {
 exports.getArticleComments = async (req, res, next) => {
   try {
     const { articleId } = req.params;
-    const associatedComments = await fetchArticleComment(articleId);
+    const validatedId = await validateArticleId(articleId);
+    const associatedComments = await fetchArticleComment(validatedId);
     res.status("200").send({ comments: associatedComments });
   } catch (err) {
     return next(err);
   }
 };
 
-exports.updateArticleComment = async (req, res, next) => {
+exports.addArticleComment = async (req, res, next) => {
   try {
     const { articleId } = req.params;
     const { username, body } = req.body;
     const allUsers = await fetchAllUsers();
-    const newComment = await addNewComment(articleId, username, body, allUsers);
+    const validatedId = await validateArticleId(articleId);
+    const newComment = await addNewComment(
+      validatedId,
+      username,
+      body,
+      allUsers
+    );
     res.status("200").send({ newComment });
   } catch (err) {
     return next(err);
   }
 };
+
+exports.updateArticle = async (req, res, next) => {
+  try {
+    const { articleId } = req.params;
+    const { inc_votes } = req.body;
+    const validatedArticleId = await validateArticleId(articleId);
```

Comment:
There shouldn't be a need to check if the article_id exists first, lets always allow psql to throw an error if it's going to and if not, see if we can make some decisions based on the response, only if we still can't do this based on the response will we need to first check if it exists. But that's generally a last resort.